polaris-react/src/components/TopBar/components/SearchField/SearchField.scss

```diff
@@ -32,15 +35,29 @@ $search-icon-width: calc(#{$icon-size} + var(--p-space-4));
   border: none;
   color: var(--p-color-text);
 
+  #{$se23} & {
+    color: var(--p-color-text-inverse);
+  }
+
   &::placeholder {
     color: var(--p-color-text-subdued);
+
+    #{$se23} & {
+      color: var(--p-color-text-inverse-subdued);
+    }
   }
 }
 
-.Input:focus-visible {
+.Input:focus-visible:not(:active) {
```

Comment:
I tested it it and it seems to be ok! But its kinda unreadable because I would expect there to be a change. I would just make a new block called `.Input:focus-visible:not(:active) {` with your updates in it. But up to you since it seems ok